src/track/macro.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <QtCore>
```

Comment:
```suggestion
#include <memory>
#include <QtCore>
```

Reply:
could you elaborate why this is needed?

Reply:
IIRC the file uses `std::shared_ptr` which requires it.

Reply:
For reference: https://en.cppreference.com/w/cpp/memory/shared_ptr

> ## std::shared_ptr
> Defined in header `<memory>`

Reply:
https://github.com/mixxxdj/mixxx/pull/2989/checks?check_run_id=1636877691#step:17:457

Reply:
Sorry, this needs to go below QtCore, my fault.